benches/odoh_bench.rs

```diff
@@ -1,61 +1,66 @@
 use criterion::{black_box, criterion_group, criterion_main, Criterion};
-use hpke::Serializable;
-use odoh_rs::key_utils::*;
-use odoh_rs::protocol::*;
-
-fn generate_key_pair() -> ObliviousDoHKeyPair {
-    let ikm = "871389a8727130974e3eb3ee528d440a871389a8727130974e3eb3ee528d440a";
-    let ikm_bytes = hex::decode(ikm).unwrap();
-    let (secret_key, public_key) = derive_keypair_from_seed(&ikm_bytes);
-    let public_key_bytes = public_key.to_bytes().to_vec();
-    let odoh_public_key = ObliviousDoHConfigContents {
-        kem_id: 0x0020,
-        kdf_id: 0x0001,
-        aead_id: 0x0001,
-        public_key: public_key_bytes,
-    };
-    ObliviousDoHKeyPair {
-        private_key: secret_key,
-        public_key: odoh_public_key,
-    }
-}
+use odoh_rs::*;
+use rand::rngs::StdRng;
+use rand::SeedableRng;
 
-fn generate_query_body() -> ObliviousDoHQueryBody {
-    ObliviousDoHQueryBody::new(
-        &hex::decode("5513010000010000000000000377777706676f6f676c6503636f6d00001c0001").unwrap(),
-        None,
-    )
-}
+pub fn bench_steps(c: &mut Criterion) {
```

Comment:
I ran benches to compare the previous code vs this code, and the previous code is an order of magnitude faster:

`parse_received_query    time:   [255.23 ns 256.12 ns 257.72 ns]`: This is equivalent to `step_decrypt_query` as it parses and decrypts the query. `step_decrypt_query      time:   [92.795 us 93.415 us 94.195 us] `.

`parse_received_response time:   [305.71 ns 317.56 ns 334.49 ns]`: This is equivalent to `step_decrypt_response` as it parses and decrypts the response. `step_decrypt_response   time:   [8.5891 us 8.6432 us 8.7128 us]`